src/outputs/parthenon_hdf5.cpp

```diff
@@ -104,6 +108,13 @@ void PHDF5Output::WriteOutputFileImpl(Mesh *pm, ParameterInput *pin, SimTime *tm
   // open HDF5 file
   // Define output filename
   auto filename = GenerateFilename_(pin, tm, signal);
+  if (signal == SignalHandler::OutputSignal::none) {
+    // After file has been opened with the current number, already advance output
+    // parameters so that for restarts the file is not immediatly overwritten again.
+    // Only applies to default time-based data dumps, so that writing "now" and "final"
+    // outputs does not change the desired output numbering.
+    UpdateNextOutput_(pm, tm);
+  }
```

Comment:
This looks like it was the previous behavior, but I think that only some of the output types deal with the signal. Should all output types only update the next output time if the signal handler is `none`?

Reply:
Agreed. I'll change it. (Assuming that doesn't break tests...)

Reply:
Ah actually I think it doesn't matter, because if the signal handler isn't none, the simulation is going to end, so it doesn't matter and output signal doesn't need to be updated. The reason for the signal check in parthenon_hdf5 is because it's BEFORE the output is actually written. So I don't think adding checks to the other outputs will actually change behavior in any way. So I lean towards leaving it, as that's fewer LOC.